File: app/src/main/java/io/github/keddnyo/midoze/utils/PackageUtils.java
package io.github.keddnyo.midoze.utils;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

public class PackageUtils {
    public String getPackageVersion(Context context, String packageName) {
        try {
            PackageInfo eInfo = context.getPackageManager().getPackageInfo(packageName, 0);
            return eInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            return null;
        }
    }
}

File: app/src/main/java/io/github/keddnyo/midoze/utils/PackageUtils.kt
package io.github.keddnyo.midoze.utils

import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.net.Uri

class PackageUtils(val context: Context, private val customPackageName: String) {
    private fun getPackageInfo(): PackageInfo? {
        return try {
            context.packageManager.getPackageInfo(customPackageName, 0)
        } catch (e: PackageManager.NameNotFoundException) {
            null
        }
    }

    fun getPackageVersionName() = getPackageInfo()?.versionName
    fun getPackageVersionBuild() = getPackageInfo()?.versionCode
    fun getPackageVersionNameAndBuild() = "${getPackageVersionName()}_${getPackageVersionBuild()}"

    fun removePackage() = with(context) {
        startActivity(
            Intent(Intent.ACTION_DELETE).apply {
                data = Uri.parse("package:$customPackageName")
            }
        )
    }
}